src/utils/common.ts

```diff
@@ -261,37 +262,38 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
   const voidSigner = new VoidSigner(senderAddress, provider);
 
   // Estimate the Gas units required to submit this transaction.
-  let nativeGasCost = gasUnits ? BigNumber.from(gasUnits) : await voidSigner.estimateGas(unsignedTx);
+  gasUnits = gasUnits ? BigNumber.from(gasUnits) : await voidSigner.estimateGas(unsignedTx);
   let tokenGasCost: BigNumber;
 
   // OP stack is a special case; gas cost is computed by the SDK, without having to query price.
   if (chainIsOPStack(chainId)) {
     assert(isOptimismL2Provider(provider), `Unexpected provider for chain ID ${chainId}.`);
     const populatedTransaction = await voidSigner.populateTransaction({
       ...unsignedTx,
-      gasLimit: nativeGasCost, // prevents additional gas estimation call
+      gasLimit: gasUnits, // prevents additional gas estimation call
     });
     // Concurrently estimate the gas cost on L1 and L2 instead of calling
     // `provider.estimateTotalGasCost` to improve performance.
     const [l1GasCost, l2GasPrice] = await Promise.all([
       provider.estimateL1GasCost(populatedTransaction),
       gasPrice || provider.getGasPrice(),
     ]);
-    const l2GasCost = nativeGasCost.mul(l2GasPrice);
+    const l2GasCost = gasUnits.mul(l2GasPrice);
     tokenGasCost = l1GasCost.add(l2GasCost);
   } else {
     if (!gasPrice) {
       const gasPriceEstimate = await getGasPriceEstimate(provider, chainId);
       gasPrice = gasPriceEstimate.maxFeePerGas;
     }
-    tokenGasCost = nativeGasCost.mul(gasPrice);
+    tokenGasCost = gasUnits.mul(gasPrice);
   }
 
   // Scale the results by the computed multiplier.
-  nativeGasCost = nativeGasCost.mul(gasTotalMultiplier).div(fixedPointAdjustment);
+  const nativeGasCost = gasUnits.mul(gasTotalMultiplier).div(fixedPointAdjustment);
```

Comment:
@dohaki Wdyt about applying the gas multiplier back in the `relayFeeCalculator`? That's the only place this function is used anyway, and I think it'd probably minimise confusion around `gasUnits` and `nativeGasCost`.

Reply:
yea that is prob cleaner. but then for both values right? `tokenGasCost` and `nativeGasCost`